microsoft/testsuites/vm_extensions/vmsnapshot_extension.py

```diff
@@ -47,6 +56,151 @@ class VmSnapsotLinuxBVTExtension(TestSuite):
+            self._verify_vmsnapshot_extension(log, node, environment)
+            trial = 0
+            while trial < 3:
+                time.sleep(2)
+                extension_dir = self._find_extension_dir(node)
+                if extension_dir == "":
+                    trial = trial + 1
+                else:
+                    break
+
+        # installing all the required packages
+        posix_os: Posix = cast(Posix, node.os)
+        posix_os.install_packages("python2.7")
+        # install pip
+        url = "https://bootstrap.pypa.io/pip/2.7/get-pip.py"
+        args = "-o get-pip.py"
+        curl = Curl(node)
+        curl.fetch(url=url, arg=args, shell=True, execute_arg="", sudo=True)
+        node.execute(cmd="python2.7 get-pip.py", shell=True, sudo=True)
+        node.execute(cmd="python2.7 -m pip --version")
+        # install mock module
+        node.execute(cmd="python2.7 -m pip install mock", sudo=True)
+        # copy the file into the vm
+        self._copy_to_node(node, "handle.txt")
+
+        if extension_dir != "":
+            # moving the handle.py file to extension directory
+            script = (
+                "#!/bin/sh\n"
+                f"mv {node.working_path}/handle.txt {extension_dir}/handle.py"
+            )
+            script_base64 = base64.b64encode(bytes(script, "utf-8")).decode("utf-8")
+            settings = {"script": script_base64}
+            extension = node.features[AzureExtension]
+            extension.create_or_update(
+                name="CustomScript",
+                publisher="Microsoft.Azure.Extensions",
+                type_="CustomScript",
+                type_handler_version="2.1",
+                auto_upgrade_minor_version=True,
+                settings=settings,
+            )
+            log.info(extension_dir)
+            # give the execution permissions to the file
+            file_path = f"{extension_dir}/handle.py"
+            permissions = "777"
+            node.tools[Chmod].chmod(path=file_path, permission=permissions, sudo=True)
+            # execute the file
+            script_result = node.execute(
+                cmd=f"python2.7 {extension_dir}/handle.py", shell=True, sudo=True
+            )
+            log.info(script_result.stdout)
+            if "True" in script_result.stdout:
+                # isSizeComputationFailed flag is set to True.
+                result.information["distro_supported_for_selective_billing"] = False
+                log.info(
+                    "unsupported distro as it do not have few of the modules "
+                    "like lsblk, lsscsi not pre-installed"
+                )
+            else:
+                result.information["distro_supported_for_selective_billing"] = True
+                log.info("supported distro")
+        else:
+            log.info("failed to find the extension directory")
+        assert extension_dir != "", "Unable to find the extension directory."
```

Comment:
use `assert extension_dir, "..."` is enough for empty checking. And this assertion can be checked before `if extension_dir != "":`. After that, the `if` is not necessary.